### PR TITLE
Migrate syslog/v2 to syslog/v3 (adds RFC3164 support to inputs.syslog) (#4593)

### LINKED ISSUE #4593: Add support for BSD style syslog messages RFC 3164 to syslog input
### Relevant telegraf.conf:

=udp4://:6514

### System info:

1.7.3

### Steps to reproduce:

1. Setup UDP input syslog plugin
2. Setup nginx output in that UDP port

### Expected behavior:

Message store in InfluxDB

### Actual behavior:

Error in plugin [inputs.syslog]: read udp4 0.0.0.0:6514: i/o timeout

### Additional info:

No idea.

### PR DESCRIPTION
**Pull request by thepacketgeek**
# Add RFC3164 Support for inputs.syslog  (#4593)

This PR replaces `go-syslog/v2` usage with `go-syslog/v3` so that the `inputs.syslog` plugin can use the [new RFC3164 support](https://github.com/influxdata/go-syslog/pull/27). Resolves #4593 

## Usage
When defining the inputs, a user will specify the `syslog_standard` option (Either "5424" *[default]* or "3164"). The `best_effort` option will also apply when "3164" is used.

## Notes
I was really hoping that this would let me use telegraf direction for Cisco syslog messages, but from my understanding it seems they have some problems with current go-syslog parsing:

Format: `<PRI>SEQNUM:HOST:MONTHDAY YEARHOUR:MINUTES:SECONDS.MILLISECONDSTIMEZONE:%APPNAME-SEVERITY-MSGID:%TAGS:MESSAGE`

Example: `<187>37972: Nov 21 16:53:33.429: %LINK-3-UPDOWN: Interface GigabitEthernet0/8, changed state to down`

- Seq field instead of version number
- No hostname provided
- Severity level, app, and MsgId is encoded in the `%APPNAME-SEVERITY-MSGID` format

@goller Is handing these messages something that might be in scope for the RFC3164 parser in go-syslog?

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Comment by thepacketgeek** [comment]
This also addresses #7023, although @leodido I'd love feedback on what you think about options to expose for RFC3164 parsing?

**Comment by endersonmaia** [comment]
Is there anything I can do to make this available on the next release ?

**Comment by thepacketgeek** [comment]
@endersonmaia I've just rebased and things should be good to go, I'm not sure how to get in front of a reviewer

**Comment by endersonmaia** [comment]
> Looks like new artifacts were built from this PR. Get them here!

wow, that's nice, I'll give it a shot and test for real on some of my devices, and give some feedback if something breaks

@thepacketgeek I'll ping some contributors here that I have already interacted, hope that's fine

/cc @danielnelson , @ssoroka

**Comment by thepacketgeek** [comment]
Thank you!!

**Comment by sjwang90** [comment]
Thanks @thepacketgeek and @endersonmaia. You both were able to test with the built artifacts? We'll get this reviewed by some of our Telegraf maintainers soon.

**Comment by endersonmaia** [comment]
I'm planning to test later this week, and will update here

Em qua, 23 de jun de 2021 15:53, Samantha Wang ***@***.***>
escreveu:

**Comment by endersonmaia** [comment]
I just tested it with one of my devices, and it's working just fine!

**Comment by thepacketgeek** [comment]
@leodido Are there any action items for me to complete, or is this PR just waiting for merge?

File: plugins/inputs/syslog/README.md
Comment (Hipska) [inline plugins/inputs/syslog/README.md:61]: Why not let the options be `RFC5424` and `RFC3164`? That makes more sense instead of just the numbers..

**Review by Hipska — APPROVED** [review]

File: plugins/inputs/syslog/README.md
Comment (thepacketgeek) [reply plugins/inputs/syslog/README.md:61]: Agreed, that is more clear :) I'll update shortly, thank you!

**Review by MyaLongmire — APPROVED** [review]

**Comment by leodido** [comment]
🔥

**Pull request merged**
